language/vm/src/proptest_types/constants.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::file_format::{Constant, SignatureToken};
+use libra_types::account_address::AccountAddress;
+use proptest::{
+    arbitrary::any,
+    collection::{btree_set, vec, SizeRange},
+    strategy::Strategy,
+};
+
+// A `ConstantPoolGen` gives you back a `Strategy` that makes a constant pool with
+// `Address` and `Vector<U8>` only.
+// TODO: make it more general and plug in a constant API
+#[derive(Clone, Debug)]
+pub struct ConstantPoolGen {
+    addresses: Vec<AccountAddress>,
+    byte_arrays: Vec<Vec<u8>>,
+}
+
+impl ConstantPoolGen {
+    // Return a `Strategy` that builds a ConstantPool with a number of `addresse`s and
+    // `vector<u8>` respectively driven by `address_count` and `byte_array_count`
+    pub fn strategy(
+        address_count: impl Into<SizeRange>,
+        byte_array_count: impl Into<SizeRange>,
+    ) -> impl Strategy<Value = Self> {
+        // get unique sets of addresses and vector<U8> (empty vector allowed)
+        (
+            btree_set(any::<AccountAddress>(), address_count),
+            btree_set(vec(any::<u8>(), 0..=20), byte_array_count),
+        )
+            .prop_map(|(addresses, byte_arrays)| Self {
+                addresses: addresses.into_iter().collect(),
+                byte_arrays: byte_arrays.into_iter().collect(),
+            })
+    }
+
+    // Return the `ConstantPool`
+    pub fn constant_pool(self) -> Vec<Constant> {
```

Comment:
Plans for more complicated values?

Reply:
well yeah, if no one else gets to it any time soon I will add it.
This is pretty self contained and it could be fun for anybody to add a complete constant pool story.
I just wanted to bring this back for now and move on to add all the other missing parts to proptest.
Also as in the comments this could benefit from a story/API around constants and I think @runtian-zhou is building one